challenger/index.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+// const level = require('level');
+const ethers = require('ethers');
+const BigNumber = require('bignumber.js');
+const { ExecutionPoker, executionId } = require('./ExecutionPoker');
+
+const cliArgs = require('./cliArgs');
+
+let Enforcer;
+let Verifier;
+try {
+  Enforcer = require('../build/contracts/Enforcer.json');
+  Verifier = require('../build/contracts/Verifier.json');
+} catch (e) {
+  console.error('Please run `npm run compile:contracts` first. 😉');
+  process.exit(1);
+}
+
+const fromWei = (wei) => {
+  const dec = new BigNumber(10).pow(18);
+  return new BigNumber(wei).div(dec).toString();
+};
+
+class MyExecutionPoker extends ExecutionPoker {
+  constructor (db, ...args) {
+    super(...args);
+    this.db = db;
+    // this.restoreSolutions();
```

Comment:
If we don't use storage, we can delete that part?
Though, what about storing the block height of the last processed event and storing that?

We can use that to replay events from that last known height on restart 🤔

Reply:
Agree on removing. Replay sounds like a good idea, but not really sure if it should be in the scope of this bounty. Up to @johannbarbie

Reply:
out of scope, but nice to have. if some-one cares to garden it, plz do